internal/metrics/metricsManager.go

```diff
@@ -119,16 +119,16 @@ type MetricsManager struct {
 	serviceLogicFilesystemUsagePercentage              *prometheus.GaugeVec
 	serviceLogicMonitoredHosts                         *prometheus.GaugeVec
 	serviceLogicMonitoredClusters                      *prometheus.GaugeVec
+	directoryUsageCollectors                           []*directoryUsageCollector
 }
 
 var _ API = &MetricsManager{}
 
-func NewMetricsManager(registry prometheus.Registerer, eventsHandler eventsapi.Handler) *MetricsManager {
+func NewMetricsManager(registry prometheus.Registerer, eventsHandler eventsapi.Handler, diskStatsHelper DiskStatsHelper, directoryUsageMonitorDirs ...string) *MetricsManager {
```

Comment:
Instead of a variadic function, I think it'd be best to have an options parameter, where we could have a list of dirs in it.